use-cases/lwa-rotation/README.md

```diff
@@ -1,219 +1,71 @@
-
-### 2. Configure Sample Solution App's IAM user
-#### I. Create IAM policy
-In order to execute the deployment script, an IAM user with the appropriate permissions is needed.
-To create a new IAM policy with the required permissions, follow the steps below.
-1. Open the [AWS console](https://console.aws.amazon.com/)
-2. Navigate to [IAM Policies console](https://us-east-1.console.aws.amazon.com/iamv2/home#/policies)
-3. Click **Create policy**
-4. Next to **Policy editor**, select **JSON** and replace the default policy with the JSON below
-```
-{
-	"Version": "2012-10-17",
-	"Statement": [
-		{
-			"Sid": "SPAPIAppIAMPolicy",
-			"Effect": "Allow",
-			"Action": [
-				"iam:CreateUser",
-				"iam:DeleteUser",
-				"iam:CreatePolicy",
-				"iam:DeletePolicy",
-				"iam:AttachUserPolicy",
-				"iam:DetachUserPolicy",
-				"iam:CreateAccessKey",
-				"iam:DeleteAccessKey"
-			],
-			"Resource": [
-				"arn:aws:iam::610134619817:user/*",
-				"arn:aws:iam::610134619817:policy/*"
-			]
-		}
-	]
+If you want to test the proposed solution, do changes and/or integrate it to your own product, follow the steps under [Deployment Guide](docs/DEPLOYMENT.md).
+
+## Workflow
+The LWA rotation workflow starts by receiving [APPLICATION_OAUTH_CLIENT_SECRET_EXPIRY](https://developer-docs.amazon.com/sp-api/docs/application-management-api-v2023-11-30-use-case-guide#step-2-register-your-sqs-queue-to-receive-expiring-credential-notifications) notification in your SQS queue. Next step is to call the Application Management API to rotate the client secret. Finally, the new secret is sent as [APPLICATION_OAUTH_CLIENT_NEW_SECRET](https://developer-docs.amazon.com/sp-api/docs/application-management-api-v2023-11-30-use-case-guide#step-3-get-the-latest-client-secret-from-your-sqs-queue) notification to your SQS queue.
+
+![LWA Rotation Workflow](docs/images/lwa-rotation-workflow.png)
+
+## Sample Code
+Below you can find sample code for the most relevant LWA Rotation operations:
+* [Check SQS messages for client secret expiry notification](#check-sqs-messages-for-client-secret-expiry-notification)
+* [Rotate client secrets](#rotate-client-secrets)
+* [Check SQS messages for new client secret notification](#check-sqs-messages-for-new-client-secret-notification)
+
+### Check SQS messages for client secret expiry notification
+
+#### Java
+*Find the full code [here](https://github.com/amzn/selling-partner-api-samples/blob/main/use-cases/lwa-rotation/code/java/src/main/java/lambda/RotateSecretsHandler.java)*
+```java
+public String handleRequest(SQSEvent event, Context context) {
+    for (SQSEvent.SQSMessage message : event.getRecords()) {
+        NotificationPayload notificationPayload = mapNotification(message.getBody());
+        if("APPLICATION_OAUTH_CLIENT_SECRET_EXPIRY".equals(notificationPayload.getNotificationType())) {
+            // proceed with additional checks and client secret rotation
+        }
+    }
```

Comment:
From our experience, many developers won't go to the full code page but just stick to the code snippets in the README. I agree that adding everything is not ideal because the code block would become too long, but I would add the steps for client Id validation and secret rotation as they are a key part of the use case.